diktat-api/src/main/kotlin/org/cqfn/diktat/api/DiktatRuleConfig.kt

```diff
@@ -0,0 +1,19 @@
+package org.cqfn.diktat.api
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Configuration of individual [DiktatRule]
+ *
+ * @property name name of the rule
+ * @property enabled
+ * @property configuration a map of strings with configuration options
+ * @property ignoreAnnotated if a code block is marked with these annotations - it will not be checked by this rule
+ */
+@Serializable
+data class DiktatRuleConfig(
+    val name: String,
+    val enabled: Boolean = true,
+    val configuration: Map<String, String> = emptyMap(),
```

Comment:
Map<string, string> will require some custom deserializers in most cases, so anyway we will need schema for that, not just a map.

Reply:
current idea is create something like
```kotlin
data class DiktatRuleConfig<C : Any>(
    val name: String,
    val enabled: Boolean = true,
    val configuration: C,
    val ignoreAnnotated: Set<String> = emptySet().
```

And parse `typealias DiktatRuleConfigRaw = DiktatRuleConfig<YamlNode>` and then parse `DiktatRuleConfigRaw` to a specific `DiktatRuleConfig<C>` taking `C` from a specific rule...

Reply:
Good by for now, anyway need to go step-by-step here. I think we will need a fixed schema with keys that would be rule names, for example:
```
[LONG_LINE]
    enabled=true
``` 

or 

```
"LONG_LINE": {
     "enabled": "true"
}
```

Reply:
good point, and we can make such big change in scope of `v2`
will extract to a separate issue